experimental/streaming/exporter/reader/format/format.go

```diff
@@ -63,49 +67,46 @@ func AppendEvent(buf *strings.Builder, data reader.Event) {
 			buf.WriteString(" >")
 		}
 
-	case reader.FINISH_SPAN:
+	case observer.FINISH_SPAN:
 		buf.WriteString("finish ")
 		buf.WriteString(data.Name)
 
 		buf.WriteString(" (")
 		buf.WriteString(data.Duration.String())
 		buf.WriteString(")")
 
-	case reader.ADD_EVENT:
+	case observer.ADD_EVENT:
 		buf.WriteString("event: ")
 		buf.WriteString(data.Message)
 		buf.WriteString(" (")
 		data.Attributes.Foreach(func(kv core.KeyValue) bool {
-			buf.WriteString(" " + kv.Key.Variable.Name + "=" + kv.Value.Emit())
+			buf.WriteString(" ")
+			buf.WriteString(kv.Key.Variable.Name)
+			buf.WriteString("=")
+			buf.WriteString(kv.Value.Emit())
 			return true
 		})
 		buf.WriteString(")")
 
-	case reader.MODIFY_ATTR:
-		buf.WriteString("modify attr")
-	case reader.RECORD_STATS:
-		buf.WriteString("record")
-
-		for _, s := range data.Stats {
-			f(false)(core.Key{
-				Variable: s.Measure.V(),
-			}.Float64(s.Value))
-
-			buf.WriteString(" {")
-			i := 0
-			s.Tags.Foreach(func(kv core.KeyValue) bool {
-				if i != 0 {
-					buf.WriteString(",")
-				}
-				i++
-				buf.WriteString(kv.Key.Variable.Name)
-				buf.WriteString("=")
-				buf.WriteString(kv.Value.Emit())
-				return true
-			})
-			buf.WriteString("}")
+	case observer.MODIFY_ATTR:
+		buf.WriteString(data.Type.String())
+
+	case observer.UPDATE_METRIC, observer.SET_METRIC:
+		formatMetricUpdate(buf, data.Type, data.Measurement)
+
+	case observer.BATCH_METRIC:
+		buf.WriteString("BATCH ")
+		for _, m := range data.Measurements {
+			// TODO: This breaks down because Gauge supports more than
```

Comment:
Looks like `Measurement` is ambiguous in this case. Just an idea: should `metric.Measurement` should be an interface and gets returned by the implementations of the `Float64Gauge` and others interfaces? So the `Float64Gauge` interface would have two more functions like:

```go
SetMeasurement(ctx context.Context, value float64, labels ...core.KeyValue) Measurement
AddMeasurement(ctx context.Context, value float64, labels ...core.KeyValue) Measurement
```

or we can keep `Measurement` as a concrete type (we seem to have a problem with interfacing all the things) and then add another enum:

```go
type MeasurementKind int

const (
	F64GaugeAdd MeasurementKind = iota
	F64GaugeSet
	F64CumulativeInc
	F64MeasureRecord
	… // same for I64
)
```

and `Measurement` would have an additional `Kind` field. Then we wouldn't need to add anything to the `Float64Gauge` interface and others.

Reply:
With that change, maybe we won't need both `SET_METRIC` and `UPDATE_METRIC`. That information would already be a part of the `Measurement` object.

Reply:
Done (and agreed in our working session).

Reply:
I forgot to add that there was some discussion about this, and a decision to support only MeasureMetric updates in the RecordBatch API.  As it stands in the PR, any metric could be passed, which I actually prefer, but it's something we could change later.